src/unittest/python/plugins/python/pylint_plugin_tests.py

```diff
@@ -15,45 +15,81 @@
+        self._run_execute_tool().assert_called_with(self.project, "pylint", ["pylint", "--test", "-f", "--x=y"], True)
 
-        expected_command_line = ('pylint',)
-        mock_assert_can_execute.assert_called_with(expected_command_line, 'pylint', 'plugin python.pylint')
+    def test_should_show_error_message_in_pyb_logs(self):
+        self._make_temp_test_file('E:476, 8: Error message')
+        self._run_execute_tool()
+        self.mock_logger.error.assert_called_with('Pylint: Module : E:476, 8: Error message')
 
-    @patch('pybuilder.plugins.python.pylint_plugin.execute_tool_on_modules')
-    def test_should_run_pylint_with_default_options(self, execute_tool):
-        project = Project(".")
-        init_pylint(project)
+    def test_should_show_warning_message_in_pyb_logs(self):
+        self._make_temp_test_file('W:2476, 8: Warning message')
+        self.project.set_property('pylint_show_warning_messages', True)
+        self._run_execute_tool()
+        self.mock_logger.warn.assert_called_with('Pylint: Module : W:2476, 8: Warning message')
 
-        execute_pylint(project, Mock(Logger))
+    def test_should_break_build_on_errors(self):
+        self._make_temp_test_file('E:2476, 8: Warning message')
+        self.project.set_property('pylint_break_build_on_errors', True)
+        with self.assertRaises(BuildFailedException):
+            self._run_execute_tool()
 
-        execute_tool.assert_called_with(project, "pylint", ["pylint"] + DEFAULT_PYLINT_OPTIONS, True)
+    def test_should_break_build_on_too_low_pylint_score(self):
+        self._make_temp_test_file('Your code has been rated at 4.60/10 (previous run: 4.60/10, +0.00)')
+        self.project.set_property('pylint_score_threshold', 5.0)
+        with self.assertRaises(BuildFailedException):
+            self._run_execute_tool()
 
-    @patch('pybuilder.plugins.python.pylint_plugin.execute_tool_on_modules')
-    def test_should_run_pylint_with_custom_options(self, execute_tool):
-        project = Project(".")
-        init_pylint(project)
-        project.set_property("pylint_options", ["--test", "-f", "--x=y"])
+    def test_should_break_build_on_too_high_pylint_score_decrease(self):
+        self._make_temp_test_file('Your code has been rated at 4.60/10 (previous run: 6.60/10, -2.00)')
+        self.project.set_property('pylint_score_change_threshold', -1.0)
+        with self.assertRaises(BuildFailedException):
+            self._run_execute_tool()
 
-        execute_pylint(project, Mock(Logger))
+    @staticmethod
+    def _make_temp_test_file(test_file_content):
+        temp_file = NamedTemporaryFile(delete=False)
+        temp_file.name = 'test.txt'
+        with open('test.txt', 'w') as f:
+            f.write(test_file_content)
 
-        execute_tool.assert_called_with(project, "pylint", ["pylint", "--test", "-f", "--x=y"], True)
+    @patch('pybuilder.plugins.python.pylint_plugin.execute_tool_on_modules')
```

Comment:
this is not a good idea, I will tell you why, please use @patch on class level like in my presentation, on slide 42